src/pages/index.js

```diff
@@ -14,70 +14,176 @@ import {
   inputJob,
 } from "../utils/constants.js";
 import "./index.css";
+import Api from "../components/Api.js";
+import PopupWithVerify from "../components/PopupWithVerify.js";
 
+//Api
+const api = new Api("https://around-api.en.tripleten-services.com/v1", {
+  authorization: "57aa7463-f839-44ea-835b-493cff8bea63",
+  "Content-Type": "application/json",
+});
+
+api
+  .getInitialCards()
+  .then((res) => {
+    res.forEach((card) => {
+      addCard(card);
+    });
+  })
+  .catch(console.error);
+
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo({
+      name: res.name,
+      job: res.about,
+    });
+    userInfo.setUserAvatar(res.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+//functions
+const handleAddSubmit = (data) => {
+  addFormPopup.handleLoadingText(true);
+  api
+    .addNewCard(data.name, data.link)
+    .then((cardData) => {
+      addCard(cardData);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      addFormPopup.handleLoadingText(false);
+    });
+};
+
+const handleEditSubmit = (inputValue) => {
+  editFormPopup.handleLoadingText(true);
+  api
+    .editProfile(inputValue.title, inputValue.description)
+    .then((data) => {
+      const profileInfo = {};
+      profileInfo.name = data.name;
+      profileInfo.job = data.about;
+      userInfo.setUserInfo(profileInfo);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      editFormPopup.handleLoadingText(false);
+    });
+};
+
+const handleDeleteButton = (card) => {
+  deletePopup.open();
+
+  deletePopup.setHandleDeleteMethod(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        card.removeCardElement();
+        deletePopup.close();
+      })
+      .catch(console.error);
+  });
+};
+
+const handleAvatarSubmit = (data) => {
+  api.updateAvatar(data.avatar).then((res) => {
+    userInfo.setUserAvatar(res.avatar);
+  });
+};
+
+const handleLikeButton = (card) => {
+  if (!card.isLiked) {
+    api
+      .addLikes(card._id)
+      .then(() => {
+        card.isLiked = !card.isLiked;
```

Comment:
Needs correction: please don't change the class variables directly, you should create a setter like I showed in the Card class